Lombiq.Hosting.Azure.ApplicationInsights/Services/TelemetryFilter.cs

```diff
@@ -23,37 +24,25 @@ public class TelemetryFilter : ITelemetryProcessor
 
     public void Process(ITelemetry item)
     {
-        if (!ShouldSend(item)) { return; }
+        if (ShouldNotSend(item)) { return; }
 
         _next.Process(item);
     }
 
-    private static bool ShouldSend(ITelemetry item)
+    private static bool ShouldNotSend(ITelemetry item)
     {
         var dependency = item as DependencyTelemetry;
-        if (dependency is not { Success: false }) return true;
+        if (dependency is not { Success: false }) return false;
 
-        dependency.Properties.TryGetValue("Error", out var dependencyError);
-        dependency.Properties.TryGetValue("Exception", out var exception);
-        dependency.Properties.TryGetValue("OrchardCore.ShellName", out var shellName);
-        var hasError = Errors.Exists(error => dependencyError?.StartsWithOrdinalIgnoreCase(error) == true) ||
-            Errors.Exists(error => exception?.StartsWithOrdinalIgnoreCase(error) == true) ||
-            exception?.Contains(GetTenantSqlErrorMessage(shellName)) == true;
-        var shouldSend = !hasError;
-
-        if (!shouldSend)
+        if (dependency.ResultCode == "409" && dependency.Name == "BlobContainerClient.Create")
         {
-            return false;
+            return true;
         }
 
-        dependency.Properties.TryGetValue("OrchardCore.DataProtectionContainerName", out var dataProtectionContainerName);
-        dependency.Properties.TryGetValue("OrchardCore.MediaBlobStorageContainerName", out var mediaBlobStorageContainerName);
-        shouldSend = !(dependency.Name == "PUT " + dataProtectionContainerName ||
-            dependency.Name == "PUT " + mediaBlobStorageContainerName);
+        dependency.Properties.TryGetValue("Error", out var error);
+        dependency.Properties.TryGetValue("Exception", out var exception);
 
-        return shouldSend;
+        return _expectedErrorsAsRegexPattern.Exists(expectedError => error?.RegexIsMatch(expectedError) == true) ||
+            _expectedErrorsAsRegexPattern.Exists(expectedError => exception?.RegexIsMatch(expectedError) == true);
```

Comment:
Since these run on every single `DependencyTelemetry`, they should consume minuscule resources. Store the whole regexes, and compiled ones, statically instead.